src/wallet/wallet.cpp

```diff
@@ -1333,7 +1333,7 @@ void CWallet::ReacceptWalletTransactions()
     {
         CWalletTx& wtx = *(item.second);
 
-        wtx.AcceptToMemoryPool(false);
+        wtx.AcceptToMemoryPool(LimitFree());
```

Comment:
I thought that the idea was to accept your own wallet transactions regardless of your relay policy.  

I can see how it may be a good idea to warn if a new transaction breaks your own relay rules, but in this case we'd be blocking an old transaction.  I don't think that this makes sense.  If we won't put it in our own mempool it'll never be relayed and therefore end up as a zombie tx forever.  It might be hard from a UX perspective to clean them up here though -- it'll look like tx just magically disappear (and they might still exist in the network).  I think that its best to accept all wallet tx into our own mempool and then have the user clean up ones that will never confirm separately.

Reply:
yes, good point.  If the user had sent a lot of free transactions and then restarted their node...then they would be subject here to rate limiting and may or may not get accepted to the mempool.  if for instance they had sent over the course of many hours enough free transactions > 150KB then some of them would not get into the mempool.  So yes you're right we should accept these into the pool regardless of how many there are.